src/crimson/os/seastore/seastore.cc

```diff
@@ -760,6 +769,30 @@ SeaStore::tm_ret SeaStore::_setattrs(
 {
   logger().debug("{} onode={}",
                 __func__, *onode);
+  auto layout = onode->get_mutable_layout(*ctx.transaction);
+  auto oi_attr = aset.find(OI_ATTR);
+  auto ss_attr = aset.find(SS_ATTR);
+  if (oi_attr != aset.end()) {
+    if (__builtin_expect(
+        oi_attr->second.length() > onode_layout_t::MAX_OI_LENGTH,
+        false)) {
+      logger().error("{} onode={} oi attr too long!");
+      return crimson::ct_error::input_output_error::make();
+    }
+    layout.oi_size = oi_attr->second.length();
+    oi_attr->second.copy_out(0, oi_attr->second.length(), &layout.oi[0]);
+  }
+
+  if (ss_attr != aset.end()) {
+    if (__builtin_expect(
+        ss_attr->second.length() > onode_layout_t::MAX_SS_LENGTH,
+        false)) {
+      logger().error("{} onode={} ss attr too long!");
+      return crimson::ct_error::input_output_error::make();
+    }
+    layout.oi_size = ss_attr->second.length();
+    ss_attr->second.copy_out(0, ss_attr->second.length(), &layout.oi[0]);
+  }
```

Comment:
Return an error if there are any other xattrs specified -- other xattrs are *legal*, we just aren't dealing with them in this PR.

Reply:
Or actually, just assert since the next step will be to handle that case.